include/arch/arm/arch/32/mode/fastpath/fastpath.h

```diff
@@ -128,7 +128,7 @@ static inline void NORETURN FORCE_INLINE fastpath_restore(word_t badge, word_t m
 
     register word_t badge_reg asm("r0") = badge;
     register word_t msgInfo_reg asm("r1") = msgInfo;
-    register word_t cur_thread_reg asm("r2") = (word_t)cur_thread;
+    register word_t cur_thread_reg asm("r2") = (word_t)cur_thread->tcbArch.tcbContext.registers;
```

Comment:
@lsf37: How come this commit doesn't trigger a preprocessor fail for armv7?

Reply:
Both functions have a `/** DONT_TRANSLATE */` tag, so the AST nodes in the parser will have no content.